sdk/storage/Azure.Storage.Blobs/README.md

```diff
@@ -33,77 +44,131 @@ Blob storage is designed for:
+
+// Open a file and upload it's data
+using (FileStream file = File.OpenRead("local-file.jpg"))
 {
-    await blob.UploadAsync(data);
+    blob.Upload(file);
 }
 ```
 
 ### Downloading a blob
+
 ```c#
-string connectionString = <connection_string>;
-var service = new BlobServiceClient(connectionString);
-var container = service.GetBlobContainerClient("mycontainer");
-var blob = container.GetBlockBlobClient("myblob");
-
-var download = await blob.DownloadAsync();
-using (var file = File.Create("BlockDestination.txt"))
-{   
-    await download.Value.Content.CopyToAsync(file);
+string connectionString = "<connection_string>";
+
+// Get a reference to a blob named "sample-file" in a container named "sample-container"
+BlobClient blob = new BlobClient(connectionString, "sample-container", "sample-file");
+
+// Download the blob
+BlobDownloadInfo download = blob.Download();
+using (FileStream file = File.OpenWrite("downloaded-file.jpg"))
+{
+    download.Content.CopyTo(file);
 }
 ```
 
 ### Enumerating blobs
+
 ```c#
-string connectionString = <connection_string>;
-var service = new BlobServiceClient(connectionString);
-var container = service.GetBlobContainerClient("mycontainer");
+string connectionString = "<connection_string>";
 
-string marker;
-do
+// Get a reference to a container named "sample-container"
+BlobContainerClient container = new BlobContainerClient(connectionString, "sample-container");
+
+// List all of its blobs
+foreach (BlobItem blob in container.GetBlobs())
 {
-    var response = await container.ListBlobsFlatSegmentAsync(marker);
-    foreach (var blob in response.Value.BlobItems)
-    {
-        Console.WriteLine(blob.Name);
-    }
-    marker = response.Value.NextMarker;
+    Console.WriteLine(blob.Name);
 }
-while (!string.IsNullOrEmpty(marker));
 ```
 
+### Authenticating with Azure.Identity
+
+The [Azure Identity library](identity) provides easy Azure Active Directory support for authentication.
+
+```c#
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+
+// Create a BlobServiceClient that will authenticate through Active Directory
+Uri accountUri = new Uri("https://MYSTORAGEACCOUNT.blob.core.windows.net/");
+BlobServiceClient client = new BlobServiceClient(accountUri, new DefaultAzureCredential());
+```
+
+Learn more about enabling Azure Active Directory for authentication with Azure Storage in [our documentation](storage_ad) and \[our samples](COMING TODAY).
+
 ## Troubleshooting
+
 All Blob service operations will throw a
 [StorageRequestFailedException][StorageRequestFailedException] on failure with
-helpful [`ErrorCode`s][error_codes].
+helpful [`ErrorCode`s][error_codes].  Many of these errors are recoverable.
+
+```c#
+string connectionString = "<connection_string>";
+
+// Try to create a container named "sample-container"
+BlobContainerClient container = new BlobContainerClient(connectionString, Randomize("sample-container"));
+try
+{
+    container.Create();

```

Comment:
Given the confusion that has existed over this exact implementation for CreateIfNotExists, I wonder if it should be avoided, and instead use a different operation as an example, like download.

Alternatively, perhaps add a 1-line comment to the effect of avoiding race conditions that exist using a separate existance check.

Reply:
Since we don't have a `CreateIfNotExists`, I think this is going to be very common for users so I'd like to have it in the README.  I'll add a comment about avoiding race conditions with a separate check.